java/org/contikios/cooja/plugins/skins/TrafficVisualizerSkin.java

```diff
@@ -76,8 +76,8 @@ public void update(Observable obs, Object obj) {
       if (last != null && historyList.size() < MAX_HISTORY_SIZE) {
         synchronized(historyList) {
           historyList.add(new RadioConnectionArrow(last));
-          visualizer.repaint(500);
         }
+        java.awt.EventQueue.invokeLater(() -> visualizer.repaint(500));
```

Comment:
`repaint()` should be fine to call from any thread but good to move the call out from the synchronized block.

Reply:
These were detected when running with `-Dcooja.debug.repaint=true`, I think it's easier to just run them from the AWT thread to avoid the noise of these warnings.

Reply:
Interesting. Well, no harm to always run them from AWT thread.